iib/workers/tasks/build.py

```diff
@@ -785,6 +936,30 @@ def handle_add_request(
     _update_index_image_build_state(request_id, prebuild_info)
 
     with tempfile.TemporaryDirectory(prefix='iib-') as temp_dir:
+        if from_index:
+            msg = 'Checking if bundles are already present in index image'
+            log.info(msg)
+            set_request_state(request_id, 'in_progress', msg)
+            present_bundles = _get_present_bundles(from_index, temp_dir)
+            filtered_bundles = _get_missing_bundles(present_bundles, resolved_bundles)
+            excluded_bundles = [
```

Comment:
You might want to put this after the `if` block to create this variable only if necessary.

Reply:
Resolved bundles actually gets overwritten by filtered bundles the following line, so that wouldn't work. I guess I could not overwrite it and use filtered_bundles instead, but that would break the function's convention.

So I decided to keep it as is. I hope I don't get -1 for this lol

Reply:
Lol no, I don't give -1's. This works :)